### DISCUSSION TITLE
Event Specific logs/Metrics

### DISCUSSION THREAD

**Discussion by hemanthofficial3009** [vectordotdev/vector#9311] [General]
Hello,
I would like to know if anyone had tried generating or collecting the event specific information as part of the logs/metrics for an vector instance. 
Scenario: 
We have a vector instance that generates events (_sources_-Http post messages) and forwards them to another vector instance over specific network port. The messages that are received at the vector instance are then redirected/written to a specific file using the _sinks_ plugin.

I was wondering if we can have the background information such as _timestamps_, _received messages length_ and some more information that is specific to a single event at the receiver vector instance.

**Comment by jszwedko** [comment]
Hi @hemanthofficial3009 ! Sources typically add some metadata; however, the `vector` source does not add any metadata as it just transparently passes the event through. Are you using the `vector` sink/source to forward between the Vector instances?

Could you list the sources you are using and what metadata you want for each source? We could use that information to open enhancement tickets to add additional metadata where missing.

**Comment by hemanthofficial3009** [comment]
hi @jszwedko , we are using the _vector_  sink/source to forward the messages from the client and receive the same on the collector instances of vector. 
Currently we are using http, socket sources for now, probably we will need to get in more as per the requirement that come up in further development. 
We are trying to fetch the

-  time stamp for each message/event sent out from the source
-  count of messages/events sent  by the Source _vector_ 

please let me know if the input provided is clear enough or not.

thanks for responding back.

**Reply by jszwedko** [reply]
> count of messages/events sent by the Source vector

This should be reported as an internal metric: https://vector.dev/docs/reference/configuration/sources/vector/#events_out_total

> time stamp for each message/event sent out from the source

This would be new. Do you mean you want the timestamp of when the event was received by the `vector` source? We could open an enhancement ticket.

**Reply by hemanthofficial3009** [reply]
The way we are planning to use vector in the network of systems is

One system will be acting as a collector where the source will be the events coming from other instance of vector which will be written to a file (for now). before writing there is also a chance of enrichment(transforms)

Each of the other systems will have an instance of vector (for ease of understanding, lets call it as a client) running on it and will be sourcing logs from other applications running on the same system (using http/tcp/unix). These events will be forwarded to the collector(above) using "sink=vector". This system can also in future have transforms in them to enrich the events being forwarded.



In such a case, we are trying to look at the time taken by a log (say generated by an application) to traverse through each of the components(source, transform, sink)from a client to reach the collector.

<--------------Client------------------------------> <------Collector------------------>

source="http"/"tcp"/"unix"->transforms->sink="vector" -----> source="vector"->transform->sink="file"

Here the information that we looking for is

1. The entry and exit time stamp for each event/message at various phases i.e., Sources, Transforms, Sinks being sent out from the client to the collector

**Reply by jszwedko** [reply]
I see, that makes sense, thanks for explaining. We do have an issue around tracking [event latency](#3445). Does that match what you are thinking?

**Reply by hemanthofficial3009** [reply]
@jszwedko Thanks for responding back. Could you please provide the specific reference to the issue around tracking event latency, as the link provided earlier seems to be pointing to this discussion itself.

**Reply by jszwedko** [reply]
Oops, sorry about that. https://github.com/vectordotdev/vector/issues/3445 is the link.

**Reply by hemanthofficial3009** [reply]
Yeah the issue #3445 it matches exactly to what we are expecting.

**Reply by hemanthofficial3009** [reply]
@jszwedko Thanks for all the help and support, could you please let me know what is the estimated time frame by which the issue #3445 is addressed. As this feature has become a key element as part of our development.
Also I would like to know whether the _vector_ acting as a source would add any timestamp by itself to the event or would it directly forward the event/log coming in from various sources as is?

**Reply by jszwedko** [reply]
Hi @hemanthofficial3009 ! Apologies for the delay. That currently isn't on our roadmap, but I could see us possibly digging into it next year. The `vector` source just passes through the event as-is, without any modification.
